packages/@react-spectrum/breadcrumbs/src/Breadcrumbs.tsx

```diff
@@ -25,6 +25,42 @@ import {useProviderProps} from '@react-spectrum/provider';
 const MIN_VISIBLE_ITEMS = 1;
 const MAX_VISIBLE_ITEMS = 4;
 
+type useResizeObserverOptionsType<T> = {
+  ref: RefObject<T>,
+  onResize: (size: {
+    width: number,
+    height: number,
+  }) => void
+}
+
+function useResizeObserver<T extends HTMLElement>(options: useResizeObserverOptionsType<T>) {
+  const {ref, onResize} = options;
+
+  const resizeObserverInstance = new ResizeObserver((entries) => {
+    if (!entries.length) {
+      return;
+    }
+
+    const entry = entries[0]; // We just need the first element
+
+    onResize({
+      width: entry.contentRect.width,
+      height: entry.contentRect.height
+    });
+  });
+
+  useEffect(() => {
+    if (!ref) { return; }
+
+    resizeObserverInstance.observe(ref.current);
+
+    return () => resizeObserverInstance.unobserve(ref.current);
+  }, [ref]);
+
+  return {ref};
```

Comment:
out of curiosity, why return the ref?

Reply:
Initially, I thought I will return the ref from the hook and use it on the component, but then I need to handle if the ref is passed as props, so I decided to go with the ref as an argument to the hook and I forgot to remove the return statement, I will remove it :D